packages/examples/src/use-tools.tsx

```diff
@@ -22,9 +22,39 @@ function App({ query }: { query: string }) {
     },
   };
   return (
-    <UseTools showSteps tools={tools} fallback="Failed to evaluate the mathematical expression">
-      <UserMessage>{query}</UserMessage>
-    </UseTools>
+    <ShowConversation
+      present={(msg) => {
+        switch (msg.type) {
+          case 'assistant':
+            return (
+              <>
+                Assistant: {msg.element}
+                {'\n'}
+              </>
+            );
+          case 'functionCall':
+            return (
+              <>
+                Function Call: {msg.element}
+                {'\n'}
+              </>
+            );
+          case 'functionResponse':
+            return (
+              <>
+                Function Result: {msg.element}
+                {'\n'}
+              </>
+            );
+          default:
+            return null;
+        }
+      }}
+    >
```

Comment:
I like the flexibility, but the example does seem way more scary now, plus, the main logic is buried deep which might be confusing.

Reply:
Also, I haven't run this but IIRC FunctionCall and FunctionResponse already print some commentary, so you might get a result like: `Function Call: Calling function X with args Y` which is redundant. One way to make this less scary is to get a `mapping` instead of a function. Assuming people always want to do a switch, this makes it easier to handle. Plus, a map can give you some defaults (e.g. if I only want to change `FunctionResponse`) instead of having to either not use `ShowConversation` vs handling all cases.

Reply:
I started with the mapping approach but the object-of-functions seemed awkward and I didn't think the existing `<FunctionCall>`/`<FunctionResponse>` values were reasonable defaults. But maybe if we're good with those being silent by default that would make sense? WDYT?

Re: the demo:
- Showing the intermediate steps is IMO pretty informative
- I wasn't sure how to make the demo legible _without_ `<ShowConversation>` (recall that I used to have the "\n" hacks to work around this)

Reply:
I'm going to resolve/merge to be able to start consuming this elsewhere but I'm not particularly wedded to the precise API and/or what's in the example.